omniNotes/src/androidTest/java/it/feio/android/omninotes/screens/Home.java

```diff
@@ -48,4 +57,71 @@ public static void clickTextNoteButton() {
     public static boolean isReminderSetDisplayed() {
         return HelperMethods.isObjectDisplayed(withId(R.id.card_layout));
     }
+
+    public static void deletePayTheBillsNote() {
+        HelperMethods.deleteElement(withId(R.id.root));
+    }
+
+    public static boolean isArchiveButtonDisplayed() {
+        return HelperMethods.isObjectDisplayed(allOf(withId(R.id.title), withText(Constants.ARCHIVE)));
+    }
+
+    public static boolean isArchiveButtonSiblingWithNotesButton() {
+        Matcher<View> archiveButton = allOf(withChild(withChild(allOf(withId(R.id.title), withText(Constants.ARCHIVE)))), hasSibling(withChild(withChild(allOf(withId(R.id.title), withText(Constants.NOTES))))));
+        return HelperMethods.isObjectDisplayed(archiveButton);
+    }
+
+    public static void clickArchiveButton() {
+        HelperMethods.performClick(allOf(withId(R.id.title), withText(Constants.ARCHIVE)));
+    }
+
+    public static void clickTrashButton() {
+        HelperMethods.performClick(allOf(withId(R.id.title), withText(Constants.TRASH)));
+    }
+
+    public static void clickMoreOptions() {
+        HelperMethods.performClick(withContentDescription(Constants.MORE_OPTIONS));
+    }
+
+    public static void clickEmptyTrash() {
+        HelperMethods.performClick(allOf(withId(R.id.title), withText(Constants.EMPTY_TRASH)));
+    }
+
+    public static void clickOKButton() {
+        HelperMethods.performClick(withId(R.id.buttonDefaultPositive));
+    }
+
+    public static boolean isTrashButtonDisplayed() {
+        return HelperMethods.isObjectDisplayed(allOf(withId(R.id.title), withText(Constants.TRASH)));
+    }
+
+    public static int getListViewNoOfChildren() {
```

Comment:
You don't need this function. You can use directly "getListViewChildCount" function in your test.

Reply:
Yes, but then I would have to put the matcher in the test or to hardcode the method by putting the matcher directly in it.